components/automate-deployment/pkg/history/history.go

```diff
@@ -0,0 +1,108 @@
+package history
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+var nowProvider = func() time.Time {
+	return time.Now()
+}
+
+type Logger struct {
+	Tags []Tag
+}
+
+func WithTags(tags ...Tag) *Logger {
+	return &Logger{
+		Tags: tags,
+	}
+}
+
+func LogEvent(ctx context.Context, name EventClass, description string) context.Context {
```

Comment:
I wonder if `RecordEvent` might be better?

Reply:
can we call `Do` -> `Record`?